lib/Dialect/Ibis/Transforms/IbisPortrefLowering.cpp

```diff
@@ -16,13 +16,24 @@
 #include "mlir/IR/Builders.h"
 #include "mlir/Transforms/DialectConversion.h"
 #include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "ibis-lower-portrefs"
 
 using namespace mlir;
 using namespace circt;
 using namespace ibis;
 
 namespace {
 
+// Prints an operation in generic form, for debugging purposes. Generic form is
+// chosen to avoid triggering op verification.
+static void debugPrintOp(Twine msg, Operation *op) {
```

Comment:
This seems so generic, is there no more central place to put this?

Reply:
This seems like it should be temporary. @mortbopet if the verifier runs and fails, does MLIR not print the generic form?

I actually think this sort of "printf style" debugging doesn't have any place in main.

Reply:
It does, with **a lot** of clutter printed as well. If we're lucky - some ops just plain doesn't play well with the pattern rewrite infrastructure, e.g. when you're replacing a symbol-defining op with another one of said symbol. That case will throw asserts during verifications, which seems... stupid, given that that seems like a fairly standard use case. 
Will remove.